src/plugins/ieo/components/IEOCard/index.tsx

```diff
@@ -0,0 +1,252 @@
+                    {this.getContent(state)}
+                </div>
+            </div>
+        );
+    }
+
+    private handleClick = () => {
+        const { ieo } = this.props;
+
+        this.props.onClick(ieo);
+    };
+
+    private getContent = (state: string) => {
+        switch (state) {
+            case 'preparing':
+                return this.renderPreparingItem();
+            case 'ongoing':
+            case 'distributing':
+                return this.renderInProgress();
+            case 'finished':
+                return this.renderFinished();
+            default:
+                return;
+        }
+    };
+
+    private renderPreparingItem = () => {
+        const {
+            starts_at,
+            finishes_at,
+            supply,
+            currency_id,
+        } = this.props.ieo;
+
+        return (
+            <div className="pg-ieo__card-content-block">
+                <div className="pg-ieo__card-content-block__row">
+                    <span className="pg-ieo__card-content-block__text">{this.translate('page.body.ieo.profile.info.goal')}</span>
+                    <span className="pg-ieo__card-content-block__text text-bold">
+                        {supply} {currency_id && currency_id.toUpperCase()}
+                    </span>
+                </div>
+                <div className="pg-ieo__card-content-block__row">
+                    <span className="pg-ieo__card-content-block__text">{this.translate('page.body.ieo.card.start.date')}</span>
+                    <span className="pg-ieo__card-content-block__text text-bold">{localeDate(starts_at, 'fullDate')}</span>
+                </div>
+                <div className="pg-ieo__card-content-block__row">
+                    <span className="pg-ieo__card-content-block__text">{this.translate('page.body.ieo.card.end.date')}</span>
+                    <span className="pg-ieo__card-content-block__text text-bold">{localeDate(finishes_at, 'fullDate')}</span>
+                </div>
+            </div>
+        );
+    };
+
+    private renderInProgress = () => {
+        const { supply, currency_id } = this.props.ieo;
+
+        return (
+            <div className="pg-ieo__card-content-block">
+                <div className="pg-ieo__card-content-block__row">
+                    <span className="pg-ieo__card-content-block__text">{this.translate('page.body.ieo.profile.info.goal')}</span>
+                    <span className="pg-ieo__card-content-block__text text-bold">
+                        {supply} {currency_id && currency_id.toUpperCase()}
+                    </span>
+                </div>
+                {this.renderProgressBar()}
+            </div>
+        );
+    };
+
+    private renderFinished = () => {
+        const { tokens_ordered, pairs } = this.props.ieo;
+        const { currency } = this.props;
+        const amountOfQuote = tokens_ordered && currency && Decimal.format(+tokens_ordered * +pairs[0].price, currency.precision);
```

Comment:
pairs can be undefined. In this case ```pairs[0].price``` will raise an error.
Better to have default value here to avoid word ```undefined``` in render